input_name.py

```diff
@@ -0,0 +1,101 @@
+from background import Background
+from constants import *
+from button import Button
+from pygame_textinput import TextInputVisualizer, TextInputManager
+
+
+class InputName(pygame.sprite.Sprite):
+    #Constructor
+    def __init__(self):
+        super().__init__()
+        #Instruction and re-enter name strings
+        self.instruction_string = "Please input your name(must be between 1 and 15 characters) below:"
+        self.reenter_string = "Your name must be at least 1 character, please try again"
+        self.max_len = 15
+        #Set up background
+        self.image = Background(BACKGROUND_IMAGES_FILE_PATHS).image.convert()
+        self.rect = self.image.get_rect()
+        #Display the instructions for inputting a name
+        self.instruction_text = pygame.font.SysFont("arialblack", 20).render(self.instruction_string, True, 'white')
+        self.instruction_textpos = self.instruction_text.get_rect(centerx=self.rect.width / 2, centery=self.rect.height / 3)
+        self.image.blit(self.instruction_text, self.instruction_textpos)
+        #Set up re-enter string text objects
+        self.reenter_text = pygame.font.SysFont("arialblack", 16).render(self.reenter_string, True, 'red')
+        self.reenter_textpos = self.reenter_text.get_rect(centerx=self.rect.width / 2,
+                                                                  centery=self.rect.height / 1.5)
+
+        #Manager for the TextInputVisualizer, it will stop inputs from being less than 0 and greater than 15
+        self.manager = TextInputManager(validator = lambda input: len(input) <= self.max_len and len(input) > 0)
+
+        #Generate instruction_text input box
+        green_rgb = (0,145,0)
+        white_rgb = (255,255,255)
+        self.input_text_font = pygame.font.SysFont("arial", 24)
+        self.text_input_box = TextInputVisualizer(manager= self.manager, font_object=self.input_text_font,
+                                                  font_color=green_rgb, cursor_color=white_rgb)
+
+        # Single character width, used to calculated background box size behind the text input and other locations
+        self.single_char_width, self.single_char_height = self.input_text_font.size("a")
+        #make the surface height the same as the font and the width the same as the max name length (the just +2 makes the graphics look cleaner)
+        self.input_text_background = pygame.Surface((self.single_char_width * 15 + 2, self.single_char_height))
+        self.input_text_background.fill(white_rgb)
+        #put input box in the center of the screen but offset to the left by half the max length of characters
+        self.image.blit(self.input_text_background, (self.rect.width / 2 - self.single_char_width * (self.max_len / 2), self.rect.height / 2))
```

Comment:
seems like a lot of work to get the format looking good in this section

Reply:
It did take some trial and error haha